components/TextFields/examples/MultilineTextFieldLegacyExample.m

```diff
@@ -197,6 +251,15 @@ - (void)viewDidLoad {
                     [[note.userInfo valueForKey:UIKeyboardFrameEndUserInfoKey] CGRectValue];
                 self.scrollView.contentInset = UIEdgeInsetsMake(0, 0, CGRectGetHeight(frame), 0);
               }];
+  [[NSNotificationCenter defaultCenter]
+      addObserverForName:UIKeyboardWillChangeFrameNotification
+                  object:nil
+                   queue:[NSOperationQueue mainQueue]
+              usingBlock:^(NSNotification *_Nonnull note) {
+                CGRect frame =
+                    [[note.userInfo valueForKey:UIKeyboardFrameEndUserInfoKey] CGRectValue];
+                self.scrollView.contentInset = UIEdgeInsetsMake(0, 0, CGRectGetHeight(frame), 0);
```

Comment:
You may need to weakify - I believe `self` will be captured in this block.

Reply:
No. This one is fine: https://stackoverflow.com/questions/20030873/always-pass-weak-reference-of-self-into-block-in-arc

Reply:
You're right that there isn't a retain cycle here. However, the block will be copied (onto the heap) when the notification center retains it. `self` will be in scope and will be retained by the notification center. As a global singleton, the notification center will retain the example VC until the application exits.  I pulled the PR and tested and was able to confirm that the VC is leaked each time it's pushed and popped.

I believe a safe approach is to register for the notifications in a lifecycle method like `viewWillAppear` and unregister in its sibling `viewDidDisappear`. Moving to the `@selector` API wont help things on iOS 8 (but [it will help on iOS 9+](https://useyourloaf.com/blog/unregistering-nsnotificationcenter-observers-in-ios-9/)).  You would still need to handle unregistering manually.

Reply:
Ok, great. That actually makes the most sense to me. I guess the example I found in SO was using it in a block that was in local scope, not global scope.

Reply:
I __wish__ this was caught by the static analyzer.

Reply:
I changed them all to selectors and added removeObservers!